ReactNativeClient/lib/services/queryBuilder.ts

```diff
@@ -0,0 +1,71 @@
+interface Term {
```

Comment:
This file and filterParser should be moved in their own directory. Perhaps there should be a lib/services/searchengine where the SearchEngine class along with your files would go.